File: android/app/src/main/java/com/link2ur/app/viewmodel/FleaMarketViewModel.kt
package com.linku.app.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.linku.app.data.api.RetrofitClient
import com.linku.app.data.models.FleaMarketItem
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

class FleaMarketViewModel : ViewModel() {
    private val apiService = RetrofitClient.apiService
    
    private val _items = MutableStateFlow<List<FleaMarketItem>>(emptyList())
    val items: StateFlow<List<FleaMarketItem>> = _items.asStateFlow()
    
    private val _categories = MutableStateFlow<List<String>>(emptyList())
    val categories: StateFlow<List<String>> = _categories.asStateFlow()
    
    private val _isLoading = MutableStateFlow(false)
    val isLoading: StateFlow<Boolean> = _isLoading.asStateFlow()
    
    private val _errorMessage = MutableStateFlow<String?>(null)
    val errorMessage: StateFlow<String?> = _errorMessage.asStateFlow()
    
    fun loadCategories() {
        viewModelScope.launch {
            try {
                val response = apiService.getFleaMarketCategories()
                if (response.isSuccessful) {
                    _categories.value = response.body()?.categories ?: emptyList()
                }
            } catch (e: Exception) {
                // 处理错误
            }
        }
    }
    
    fun loadItems(category: String? = null, keyword: String? = null) {
        viewModelScope.launch {
            _isLoading.value = true
            _errorMessage.value = null
            
            try {
                val response = apiService.getFleaMarketItems(category = category, keyword = keyword)
                if (response.isSuccessful) {
                    _items.value = response.body()?.items ?: emptyList()
                } else {
                    _errorMessage.value = "加载失败: ${response.code()}"
                }
            } catch (e: Exception) {
                _errorMessage.value = "网络错误: ${e.message}"
            } finally {
                _isLoading.value = false
            }
        }
    }
}

File: ios/LinkU/ViewModels/FleaMarketViewModel.swift
import Foundation
import Combine

class FleaMarketViewModel: ObservableObject {
    @Published var items: [FleaMarketItem] = []
    @Published var categories: [FleaMarketCategory] = []
    @Published var isLoading = false
    @Published var errorMessage: String?
    
    private let apiService = APIService.shared
    private var cancellables = Set<AnyCancellable>()
    
    func loadCategories() {
        apiService.request(FleaMarketCategoryResponse.self, "/api/flea-market/categories", method: "GET")
            .sink(receiveCompletion: { _ in }, receiveValue: { [weak self] response in
                self?.categories = response.categories
            })
            .store(in: &cancellables)
    }
    
    func loadItems(category: String? = nil, keyword: String? = nil, page: Int = 1) {
        isLoading = true
        var endpoint = "/api/flea-market/items?page=\(page)&page_size=20&status=active"
        if let category = category {
            endpoint += "&category=\(category)"
        }
        if let keyword = keyword {
            endpoint += "&keyword=\(keyword)"
        }
        
        apiService.request(FleaMarketItemListResponse.self, endpoint, method: "GET")
            .sink(receiveCompletion: { [weak self] result in
                self?.isLoading = false
                if case .failure(let error) = result {
                    self?.errorMessage = error.localizedDescription
                }
            }, receiveValue: { [weak self] response in
                if page == 1 {
                    self?.items = response.items
                } else {
                    self?.items.append(contentsOf: response.items)
                }
            })
            .store(in: &cancellables)
    }
}

class FleaMarketDetailViewModel: ObservableObject {
    @Published var item: FleaMarketItem?
    @Published var isLoading = false
    @Published var errorMessage: String?
    
    private let apiService = APIService.shared
    private var cancellables = Set<AnyCancellable>()
    
    func loadItem(itemId: Int) {
        isLoading = true
        apiService.request(FleaMarketItem.self, "/api/flea-market/items/\(itemId)", method: "GET")
            .sink(receiveCompletion: { [weak self] result in
                self?.isLoading = false
                if case .failure(let error) = result {
                    self?.errorMessage = error.localizedDescription
                }
            }, receiveValue: { [weak self] item in
                self?.item = item
            })
            .store(in: &cancellables)
    }
    
    func directPurchase(itemId: Int, completion: @escaping (Bool) -> Void) {
        apiService.request(EmptyResponse.self, "/api/flea-market/items/\(itemId)/direct-purchase", method: "POST", body: [:])
            .sink(receiveCompletion: { result in
                if case .failure = result {
                    completion(false)
                }
            }, receiveValue: { _ in
                completion(true)
            })
            .store(in: &cancellables)
    }
    
    func requestPurchase(itemId: Int, proposedPrice: Double?, message: String?, completion: @escaping (Bool) -> Void) {
        var body: [String: Any] = [:]
        if let price = proposedPrice {
            body["proposed_price"] = price
        }
        if let message = message {
            body["message"] = message
        }
        
        apiService.request(PurchaseRequest.self, "/api/flea-market/items/\(itemId)/purchase-request", method: "POST", body: body)
            .sink(receiveCompletion: { result in
                if case .failure = result {
                    completion(false)
                }
            }, receiveValue: { _ in
                completion(true)
            })
            .store(in: &cancellables)
    }
}

